test/CMakeLists.txt

```diff
@@ -34,6 +34,10 @@ add_executable(arestest ${TESTSOURCES} ${TESTHEADERS})
 target_include_directories(arestest PRIVATE ${CMAKE_CURRENT_BINARY_DIR})
 target_link_libraries(arestest PRIVATE caresinternal gmock)
 
+IF (CARES_BUILD_CONTAINER_TESTS)
+  target_compile_definitions(arestest PRIVATE HAVE_USER_NAMESPACE HAVE_UTS_NAMESPACE)
+ENDIF ()
```

Comment:
I figured it'd be safe to assume container namespaces are supported when people explicitly pass the option on the command line.